### PR TITLE
added code analyzer

### PR DESCRIPTION
**Pull request by DevMike123**
Continuation of #19 . Uses Level 6 composer. All errors fixed. @Meldiron pr ready for merge. Fixes https://github.com/appwrite/appwrite/issues/4373
success

### CODE REVIEW

**Comment by DevMike123** [comment]
@Steven can you review the pr?

**Comment by DevMike123** [comment]
@stnguyen90 can you review the pr?

**Comment by PineappleIOnic** [comment]
Thank you so much for the PR 🤩. We're adding the `hacktoberfest-accepted` label to ensure this PR counts towards your Hacktoberfest contributions count. With that said, please stay active on this PR to address any comments once you receive a review. Happy Hacktoberfest! 🎃

**Comment by stnguyen90** [comment]
@DevMike123 looks like there are merge conflicts. Would you please resolve them?

**Comment by DevMike123** [comment]
> @DevMike123 looks like there are merge conflicts. Would you please resolve them?

@stnguyen90 done.

File: composer.json [hunk 1]
```diff
@@ -15,20 +15,22 @@
...
-        "php": ">=7.4",
+        "php": ">=8.0",
         "laravel/pint": "1.2.*"
     },
     "require-dev": {
         "phpunit/phpunit": "^9.3",
         "vimeo/psalm": "4.0.1",
-        "squizlabs/php_codesniffer": "^3.6"
+        "squizlabs/php_codesniffer": "^3.6",
+"phpstan/phpstan": "1.9.x-dev"
```
Comment (stnguyen90) [inline composer.json:34]: The formatting of this JSON is a little off. Would you please fix it?

File: src/System/System.php [hunk 2]
```diff
@@ -118,100 +109,98 @@ public static function getHostname(): string
...
+    public static function getCPUCores(): string|int
     {
         switch (self::getOS()) {
             case 'Linux':
                 $cpuinfo = file_get_contents('/proc/cpuinfo');
-                preg_match_all('/^processor/m', $cpuinfo, $matches);
-
-                return count($matches[0]);
+                if(is_string($cpuinfo)==1)
+                {preg_match_all('/^processor/m', $cpuinfo, $matches);
```
Comment (stnguyen90) [inline src/System/System.php:178]: This doesn't match our code style. Please use `composer format` to ensure it matches our style.

File: composer.json [hunk 3]
```diff
@@ -15,20 +15,22 @@
             "email": "torsten@appwrite.io"
         }
     ],
+"scripts": {
+        "check": "./vendor/bin/phpstan analyse --level 6 src tests",
```
Comment (stnguyen90) [inline composer.json:19]: level 8 only adds 2 more errors that need to be fixed. Would you be able to change to 8 and fix those errors?

**Review by stnguyen90 — CHANGES_REQUESTED** [review]
Great PR! 🤯 We left some comments during the review, please check them out.

**Pull request closed without merging**